### PR TITLE
docs: fix hyperlinks and other minor issues

### PR DESCRIPTION
**Pull request by qmonnet**
This is a fix for hyperlinks under `Documentation/`, and for other (mostly) related issue. I ran [a tool](https://github.com/dkhamsing/awesome_bot) to check links on documentation, and found a number of dead link. This tool also reported links pointing to redirections, or duplicate links in source files. Let's try to fix all of this:

- Fix broken links (404)
- Fix links pointing to wrong targets
- Use final destination for (most) links pointing to redirections (two commits, one is just for `docs.cilium.io` in file headers)
- Avoid duplicated links as much as possible, use names hyperlinks instead
- While at it, replace references to “covalent” in links, let's avoid any confusion for the readers

The STAR WARS API used in the documentation, tests and examples for TLS visibility has been taken down. Let's replace it with something else (https://artii.herokuapp.com), cc  @danwent.

Additional miscellaneous fixes:

- Fix a typo in CI doc (`ct/next-next`, sounds funny but is unused)
- Fix retrieval of current release number in Python script, due to break some day otherwise
- Add `test-me-please` to the list of keywords triggering GKE tests in CI
- Move URL and update name for `Cilium-Nightly-Tests Job`, fix link for Flannel job, in CI doc. **Please double check this commit, I am not 100% sure the change is correct.**

After this PR, I still observe **broken links for bisects on Jenkins** (CI doc). I am unsure if the links are broken or if I simply lack permission to access them, can someone please check and let me know?

### CODE REVIEW

**Comment by qmonnet** [comment]
test-me-please

**Comment by qmonnet** [comment]
test-gke

**Comment by qmonnet** [comment]
test-with-kernel

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -325,14 +325,14 @@ var _ = Describe("K8sPolicyTest", func() {
...
-			res.ExpectSuccess("Cannot connect from %q to 'https://swapi.co:443/api/planets/1/'",
+				helpers.CurlFail("--retry 5 -4 %s 'https://artii.herokuapp.com/make?text=cilium&font=univers'", "-v --cacert /cacert.pem"))
+			res.ExpectSuccess("Cannot connect from %q to 'https://artii.herokuapp.com/make?text=cilium&font=univers'",
 				appPods[helpers.App2])
 
 			res = kubectl.ExecPodCmd(
 				namespaceForTest, appPods[helpers.App2],
-				helpers.CurlFail("--retry 5 -4 %s https://swapi.co:443/api/planets/2/", "-v --cacert /cacert.pem"))
-			res.ExpectFailWithError("403 Forbidden", "Unexpected connection from %q to 'https://swapi.co:443/api/planets/2/'",
+				helpers.CurlFail("--retry 5 -4 --max-time 15 %s 'https://artii.herokuapp.com:443/fonts_list'", "-v --cacert /cacert.pem"))
```
Comment (pchaigno) [inline test/k8sT/Policies.go]: Why did you remove the `--max-time` parameter above and add it here?

File: Documentation/conf.py [hunk 2]
```diff
@@ -91,18 +91,19 @@
...
+    next_release = current_release
 githubusercontent = 'https://raw.githubusercontent.com/cilium/cilium/'
 scm_web = githubusercontent + branch
 jenkins_branch = 'https://jenkins.cilium.io/view/Cilium-v' + current_release
 archive_filename = archive_name + '.tar.gz'
 archive_link = 'https://github.com/cilium/cilium/archive/' + archive_filename
 archive_name = 'cilium-' + archive_name.strip('v')
 project_link = 'https://github.com/cilium/cilium/projects?query=is:open+' + next_release
-backport_format = 'https://github.com/cilium/cilium/pulls?q=is:open+is:pr+label:%s/' + current_release
+backport_format = 'https://github.com/cilium/cilium/pulls?q=is:open+is:pr+label:backport/%s' + current_release
```
Comment (pchaigno) [inline Documentation/conf.py]: I _think_ this is incorrect. [Those links](https://docs.cilium.io/en/latest/contributing/release/backports/#preparation) already work. And according to [the Sphinx documentation](https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html), the `%s` here is to be replaced in the reStructuredText document, not by `current_release`.

File: test/k8sT/manifests/l7-policy-TLS.yaml [hunk 3]
```diff
@@ -16,23 +16,23 @@ spec:
...
+          name: "artii-server"
       originatingTLS:
         secret:
           namespace: "default"
           name: "test-client"
       rules:
         http:
         - method: "GET"
-          path: "/api/planets/1/"
+          path: "/make??.*"
```
Comment (pchaigno) [inline test/k8sT/manifests/l7-policy-TLS.yaml]: As discussed offline, the `??` here seems incorrect.

**Review by pchaigno — CHANGES_REQUESTED** [review]
Couple nits below, but looks good overall! Thanks for taking care of all those links :link: !

File: test/k8sT/Policies.go (hunk 1 above)
Comment (qmonnet) [reply test/k8sT/Policies.go]: Uh, that was not intentional. Thanks (So I replaced the two URLs, realised that I had swapped the one supposed to work with the one supposed to fail, and swapped the lines without noticing the `--max-time` difference). I'll fix.

File: Documentation/conf.py (hunk 2 above)
Comment (qmonnet) [reply Documentation/conf.py]: Oh I missed that, thanks for checking! I'll fix.

**Comment by qmonnet** [comment]
Fixed issues pointed by Paul.
<details><summary>Incremental diff</summary>

``` diff
diff --git a/Documentation/conf.py b/Documentation/conf.py
index d646fe4535a9..bf76ab2eec43 100644
--- a/Documentation/conf.py
+++ b/Documentation/conf.py
@@ -103,7 +103,7 @@ archive_filename = archive_name + '.tar.gz'
 archive_link = 'https://github.com/cilium/cilium/archive/' + archive_filename
 archive_name = 'cilium-' + archive_name.strip('v')
 project_link = 'https://github.com/cilium/cilium/projects?query=is:open+' + next_release
-backport_format = 'https://github.com/cilium/cilium/pulls?q=is:open+is:pr+label:backport/%s' + current_release
+backport_format = 'https://github.com/cilium/cilium/pulls?q=is:open+is:pr+label:%s/' + current_release
 
 # Store variables in the epilogue so they are globally available.
 rst_epilog = """
diff --git a/test/k8sT/Policies.go b/test/k8sT/Policies.go
index b77c50a6b044..7c1b484cdcde 100644
--- a/test/k8sT/Policies.go
+++ b/test/k8sT/Policies.go
@@ -325,13 +325,13 @@ var _ = Describe("K8sPolicyTest", func() {
 
 			res = kubectl.ExecPodCmd(
 				namespaceForTest, appPods[helpers.App2],
-				helpers.CurlFail("--retry 5 -4 %s 'https://artii.herokuapp.com/make?text=cilium&font=univers'", "-v --cacert /cacert.pem"))
+				helpers.CurlFail("--retry 5 -4 --max-time 15 %s 'https://artii.herokuapp.com/make?text=cilium&font=univers'", "-v --cacert /cacert.pem"))
 			res.ExpectSuccess("Cannot connect from %q to 'https://artii.herokuapp.com/make?text=cilium&font=univers'",
 				appPods[helpers.App2])
 
 			res = kubectl.ExecPodCmd(
 				namespaceForTest, appPods[helpers.App2],
-				helpers.CurlFail("--retry 5 -4 --max-time 15 %s 'https://artii.herokuapp.com:443/fonts_list'", "-v --cacert /cacert.pem"))
+				helpers.CurlFail("--retry 5 -4 %s 'https://artii.herokuapp.com:443/fonts_list'", "-v --cacert /cacert.pem"))
 			res.ExpectFailWithError("403 Forbidden", "Unexpected connection from %q to 'https://artii.herokuapp.com:443/fonts_list'",
 				appPods[helpers.App2])
 
diff --git a/test/k8sT/manifests/l7-policy-TLS.yaml b/test/k8sT/manifests/l7-policy-TLS.yaml
index f18045d93183..20d08213f031 100644
--- a/test/k8sT/manifests/l7-policy-TLS.yaml
+++ b/test/k8sT/manifests/l7-policy-TLS.yaml
@@ -32,7 +32,7 @@ spec:
       rules:
         http:
         - method: "GET"
-          path: "/make??.*"
+          path: "^/make(\?.*)$"
           headerMatches:
           - mismatch: REPLACE
             name: "User-Agent"
```
</details>

**Comment by qmonnet** [comment]
test-me-please

**Review by pchaigno — APPROVED** [review]

**Comment by nebril** [comment]
test-gke

**Comment by qmonnet** [comment]
test-me-please

**Review by tklauser — APPROVED** [review]

**Review by tgraf — APPROVED** [review]
